### PR TITLE
Feat/toggle button comp

### PR DESCRIPTION
**Pull request by themonster2015**
## Description

Create the toggle button component.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-core/src/components/Icon/index.tsx [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
 
+export type icontype = 'solid' | 'outline';
 export type IconName = keyof typeof HeroIcons;
-interface IconProps {
+export interface IconProps {
   icon: IconName;
   styling?: string;
+  strokeColor?: string;
+  fillColor?: string;
+  iconType?: icontype;
```
Comment (josenriagu) [inline ui/design-system-core/src/components/Icon/index.tsx]: The Icon interface has changed, so you could update with next and resolve the changes

File: ui/design-system-core/src/components/Icon/index.tsx [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
-const Icon: React.FC<IconProps> = props => {
-  const PassedIcon = HeroIcons[props.icon];
+const Icon: React.FC<IconProps> = ({
+  icon,
+  styling,
+  strokeColor = 'currentColor',
+  fillColor = 'currentColor',
+  iconType = 'outline',
+}) => {
+  const PassedIcon = iconType === 'outline' ? HeroIcons[icon] : SolidHeroIcons[icon];
```
Comment (josenriagu) [inline ui/design-system-core/src/components/Icon/index.tsx]: This too, also consider the custom Icons when resolving from next 👍

File: ui/design-system-core/src/components/Toggle/index.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          disabled={disabled}
+          checked={enabled}
+          className="peer sr-only"
+          onClick={() => setEnabled(!enabled)}
+        />
+        <div className={tw(instanceButtonStyle)}></div>
+
+        {iconChecked && iconUnchecked && (
+          <Icon
+            icon={enabled ? iconChecked : iconUnchecked}
```
Comment (josenriagu) [inline ui/design-system-core/src/components/Toggle/index.tsx]: This prop has changed from `icon` to `type`

**Review by josenriagu — CHANGES_REQUESTED** [review]

File: ui/design-system-core/src/components/Icon/index.tsx (hunk 1 above)
Comment (themonster2015) [reply ui/design-system-core/src/components/Icon/index.tsx]: Thanks. I have modified it.

File: ui/design-system-core/src/components/Icon/index.tsx (hunk 2 above)
Comment (themonster2015) [reply ui/design-system-core/src/components/Icon/index.tsx]: noted

**Review by quininez — APPROVED** [review]

**Review by josenriagu — APPROVED** [review]

**Pull request merged**